### PR TITLE
Update TF pinned version in CI

### PR DESCRIPTION
**Pull request by mudit2812**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/workflows/install_deps/action.yml
Comment (mudit2812) [inline .github/workflows/install_deps/action.yml:78]: This was a temp change just to check that the correct version is installed. I will revert this before merging. Just waiting for CI to pass.

**Comment by mudit2812** [comment]
[sc-69110]

**Review by lillian542 — APPROVED** [review]

**Review by AmintorDusko — APPROVED** [review]
Thank you, @mudit2812!

**Comment by trbromley** [comment]
Thanks! Would you mind making a story (if not done already) to track the upgrade to latest TF?

**Pull request merged**
